synapse/rest/client/push_rule.py

```diff
@@ -169,7 +169,7 @@ async def on_GET(self, request: SynapseRequest, path: str) -> Tuple[int, JsonDic
             raise UnrecognizedRequestError()
 
 
-def _rule_spec_from_path(path: Sequence[str]) -> RuleSpec:
+def _rule_spec_from_path(path: List[str]) -> RuleSpec:
```

Comment:
Ordering matters here so we need to use `List[str]` (or `Tuple[str]`) instead of `StrCollection`. There's only a single caller anyway so I don't think it is a big deal to just use `List[str]` though.

Reply:
I wouldn't object to a `StrSequence = Union[Tuple[str, ...], List[str]]` in the future if we find a need to be more general. LGTM as-is though.

Reply:
Yeah, I think in this case it is fine. 🤷